### PR TITLE
[Cms] Added a getByIdentifier() method to the BlockRepository

### PR DESCRIPTION
**Pull request by enrico69**

### CODE REVIEW

**Comment by adragus-inviqa** [comment]
So you never tried loading a block via its identifier? Try it. [It already works](https://github.com/magento/magento2/blob/5640494293ed26876700bb7c5f9578132bc5b0e6/app/code/Magento/Cms/Model/ResourceModel/Block.php#L108-L112).

File: app/code/Magento/Cms/Api/BlockRepositoryInterface.php [hunk 1]
```diff
@@ -32,6 +32,16 @@ public function save(Data\BlockInterface $block);
...
 
     /**
+     * Load Block data by given Block identifier
+     *
+     * @param string $blockIdentifier
+     * @param int $storeId
+     * @return \Magento\Cms\Api\Data\BlockInterface
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
+    public function getByIdentifier($blockIdentifier, $storeId);
```
Comment (ishakhsuvarov) [inline app/code/Magento/Cms/Api/BlockRepositoryInterface.php]: Unfortunately, due to backwards compatibility policy, currently we can not add any new methods to the interfaces, specially the ones marked with the `@api` annotation. You may extract this method to a separate interface/class, if it is really needed.

**Review by ishakhsuvarov — CHANGES_REQUESTED** [review]

File: app/code/Magento/Cms/Api/BlockRepositoryInterface.php (hunk 1 above)
Comment (enrico69) [reply app/code/Magento/Cms/Api/BlockRepositoryInterface.php]: What do you suggest? Should I simply remove this part from the interface and let the implementation in the model ( `app/code/Magento/Cms/Model/ResourceModel/Block.php` ) ?

Comment (adragus-inviqa) [reply app/code/Magento/Cms/Api/BlockRepositoryInterface.php]: Actually, your PR is redundant, as `Magento\Cms\Api\BlockRepositoryInterface::getById('my-block-identifier')` already works. As is `Magento\Cms\Api\BlockRepositoryInterface::getById(12)`, too. 

The resource model I showed you is just the place where the logic decides which route to go: identifier or numeric id? Again, try it yourself.

Comment (enrico69) [reply app/code/Magento/Cms/Api/BlockRepositoryInterface.php]: Yes but there is also this point, given by the user _convenient_

> Might be useful to be aware of #6570 and #7417 in which identifier is not enough to update cms blocks by identifier when that identifier is shared across multiple stores.

**Review by enrico69 — COMMENTED** [review]
Done

**Comment by enrico69** [comment]
@ishakhsuvarov changes are completed.

**Comment by okorshenko** [comment]
Hi @enrico69 
Thank you for your contribution. Unfortunately, we can not accept this PR. I agree with the point of @adragus-inviqa . Looks like this will be duplication of the functionality.
Thank you for your efforts.

Just FYI: we are working on https://github.com/magento/magento2/pull/7417  It should resolve your issue

**Pull request closed without merging**